### PR TITLE
use node-config module

### PR DESCRIPTION
**Pull request by smithcommajoseph**
I took the liberty of implementing this in the [rendr-cli](https://github.com/technicolorenvy/rendr-cli) and think it is a nice addition to the rendr-app-template project.

By using the node-config module, we can easily define a default config from which all other environment-specific configs can inherit. While this PR is using yaml for the conf files, it is easy enough to swap out w/ .js or .json files, i chose yaml as it feels like a good sensible default.

Additionally, we are able to remove this file form the boiler-plate setup [server/env](https://github.com/airbnb/rendr-app-template/blob/master/server/lib/env.js)

### CODE REVIEW

**Comment by spikebrehm** [comment]
This is awesome, thank you. It would be very good for us not to reinvent
the wheel. Ill take a look at this in detail tomorrow.

On Saturday, July 27, 2013, Joseph Smith wrote:

> I took the liberty of implementing this in the rendr-clihttps://github.com/technicolorenvy/rendr-cliand think it is a nice addition to the rendr-app-template project.
> 
> By using the node-config module, we can easily define a default config
> from which all other environment-specific configs can inherit from. While
> this PR is using yaml for the conf files, it is easy enough to swap out w/
> .js or .json files, i chose yaml as it feels like a good sensible default.
> 
> Additionally, we are able to remove this file form the boiler-plate setup
> 
> ## server/envhttps://github.com/airbnb/rendr-app-template/blob/master/server/lib/env.js
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/technicolorenvy/rendr-app-template node-config
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/airbnb/rendr-app-template/pull/42
> Commit Summary
> - use node-config module
> 
> File Changes
> - _A_ config/default.ymlhttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-0(23)
> - _D_ config/development.jshttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-1(27)
> - _A_ config/development.ymlhttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-2(1)
> - _D_ config/production.jshttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-3(27)
> - _A_ config/production.ymlhttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-4(1)
> - _M_ index.jshttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-5(18)
> - _M_ package.jsonhttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-6(4)
> - _D_ server/lib/env.jshttps://github.com/airbnb/rendr-app-template/pull/42/files#diff-7(9)
> 
> Patch Links:
> - https://github.com/airbnb/rendr-app-template/pull/42.patch
> - https://github.com/airbnb/rendr-app-template/pull/42.diff

## 

Spike Brehm
Software Engineer
www.airbnb.com

San Francisco, CA, USA
C: (602) 828-2358
See how Airbnb works https://www.airbnb.com/info/how_it_works

File: index.js [hunk 1]
```diff
@@ -68,9 +68,17 @@ function start() {
...
 initMiddleware();
 initServer();
+/**
+ * Here we actually initialize everything and start the Express server.
+ *
+ * We have to add the middleware before we initialize the server, otherwise
+ * the 404 handler gets too greedy, and intercepts i.e. static assets.
+ */
+initMiddleware();
+initServer();
```
Comment (spikebrehm) [inline index.js]: It looks like the calls to `initMiddleware()` and `initServer()` are duplicated from above.

File: config/default.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Default config
+
+App:
```
Comment (spikebrehm) [inline config/default.yml]: Can we make this lower-case?

File: index.js (hunk 1 above)
Comment (smithcommajoseph) [inline index.js]: DERP! sorry, that's embarrassing

File: config/default.yml (hunk 2 above)
Comment (smithcommajoseph) [inline config/default.yml]: absolutely

File: config/default.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Default config
+
+app:
+  host: localhost
```
Comment (spikebrehm) [inline config/default.yml]: Last thought: we don't actually use `host` anywhere

**Comment by spikebrehm** [comment]
Can you also add `config/runtime.json` to `.gitignore`?

**Comment by spikebrehm** [comment]
:beers:

**Pull request merged**